site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md

```diff
@@ -0,0 +1,507 @@
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json/2/3
+{
+  ...
+  "permissions": ["storage", "alarms"],
+  "permissions": ["storage"],
+  "host_permissions": ["https://extension-tips.glitch.me/*"],
+}
+```
+
+The extension will fetch all the tips and pick one at random. Then save it to storage. We will create an alarm that will be triggered once a day. Alarms are not saved when you close Chrome. So we need to check if the alarm exists and create it if it doesn't.
+
+{% Label %}sw-tips.js:{% endLabel %}
+
+```js
+// Fetch tip & save in storage
+const updateTip = async () => {
+  const response = await fetch('https://extension-tips.glitch.me/tips.json');
+  const tips = await response.json();
+  const randomIndex = Math.floor(Math.random() * tips.length);
+  return chrome.storage.local.set({ tip: tips[randomIndex] });
+};
+
+const ALARM_NAME = 'tip';
+
+// Check if alarm exists to avoid resetting the timer.
+// The alarm might be removed when the browser session restarts.
+async function createAlarm() {
+  const alarm = await chrome.alarms.get(ALARM_NAME);
+  if (typeof alarm === 'undefined') {
+    chrome.alarms.create(ALARM_NAME, {
+      delayInMinutes: 1,
+      periodInMinutes: 1440
+    });
+    updateTip();
+  }
+}
+
+createAlarm();
+
+// Update tip once a the day
+chrome.alarms.onAlarm.addListener(updateTip);
+```
+
+{% Aside 'important' %}
+
+All [Chrome API][doc-apis] event listeners and methods restart the service worker's 30 second termination timer. For more information, see the [Extension service worker lifecycle](tdb).
+
+{% endAside %}
+
+### Step 7: Communicate with other contexts {: #step-7 }
+
+Extensions use [content scripts][doc-content] to read and modify the content of the page. When a user visits a Chrome API reference page, the extension's content script will [send a message][doc-messages] to request the tip of the day from the service worker. 
+
+Start by declaring the content script in the manifest and add the match pattern corresponding to the [Chrome API][doc-apis] reference documentation.
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json
+{
+  ...
+  "content_scripts": [
+    {
+      "matches": ["https://developer.chrome.com/docs/extensions/reference/*"],
+      "js": ["content.js"]
+    }
+  ]
+}
+
+```
+
+Create a new content file. The following code sends a message to the service worker requesting the tip. Then, adds a button that will open a popover containing the extension tip. This code uses the new web platform [Popover API][popover-chrome-status] (see the [HTML spec][popover-html-spec] for more details). 
+
+{% Label %}content.js:{% endLabel %}
+
+```js
+(async () => {
+  // Sends a message to the service worker and receives a tip in response
+  const { tip } = await chrome.runtime.sendMessage({ greeting: 'tip' });
```

Comment:
```suggestion
  const { tip } = await chrome.runtime.sendMessage({ type: 'tip' });
```

Reply:
Thanks for this suggestion.  Since we already merged the sample code, I'd rather leave this as is for now. If we find a critical change to the source code, I will add this to the list.